src/reanimated2/frameCallback/FrameCallbackRegistryUI.ts

```diff
@@ -14,21 +29,44 @@ export const prepareUIRegistry = runOnUI(() => {
   'worklet';
 
   const frameCallbackRegistry: FrameCallbackRegistryUI = {
-    frameCallbackRegistry: new Map<number, () => void>(),
+    frameCallbackRegistry: new Map<number, CallbackDetails>(),
     frameCallbackActive: new Set<number>(),
     isFrameCallbackRunning: false,
+    previousFrameTimestamp: undefined,
 
     runCallbacks() {
-      const loop = () => {
+      const loop = (timestamp: number) => {
+        if (this.previousFrameTimestamp === undefined) {
+          this.previousFrameTimestamp = timestamp;
+        }
+
+        const timeSinceLastFrame = timestamp - this.previousFrameTimestamp;
+
         this.frameCallbackActive.forEach((key: number) => {
-          const callback = this.frameCallbackRegistry.get(key);
-          callback!();
+          const callback = this.frameCallbackRegistry.get(key)!;
+
+          const startTime = callback.startTime;
```

Comment:
You can also use destructuring assignment:
```suggestion
          const { startTime } = callback;
```